### PR TITLE
Jetpack Onboarding: Save site type selection

### PR DESCRIPTION
**Pull request by tyxla**
This PR updates the site type step to be saved in the remote site when clicking the corresponding tile.

To test:
1. Checkout this branch on your local Calypso.
1. Make sure your JP sandbox is running https://github.com/Automattic/jetpack/pull/8431.
1. Go to https://YourJetpackSandbox.com/wp-admin/admin.php?page=jetpack&action=onboard&calypso_env=development where `YourJetpackSandbox.com` is the domain of your Jetpack sandbox.
1. After you land the JPO flow, skip to the Site Type step.
1. Click the "Personal" tile.
1. Verify the `jpo_site_type` option was saved to `personal` (to check this, go to https://YourJetpackSandbox.com/wp-admin/options.php and look for the `jpo_site_type` option)
1. Repeat steps 5-6 again, but this time with the Business tile, and `business` as an option value.

### CODE REVIEW

File: client/jetpack-onboarding/steps/site-type.jsx [hunk 1]
```diff
@@ -15,12 +16,24 @@ import PageViewTracker from 'lib/analytics/page-view-tracker';
 import Tile from 'components/tile-grid/tile';
 import TileGrid from 'components/tile-grid';
 import { JETPACK_ONBOARDING_STEPS as STEPS } from '../constants';
+import { saveJetpackOnboardingSettings } from 'state/jetpack-onboarding/actions';
 
 class JetpackOnboardingSiteTypeStep extends React.PureComponent {
+	handleSiteTypeSelection = siteType => {
+		const { siteId } = this.props;
```
Comment (ockham) [inline client/jetpack-onboarding/steps/site-type.jsx]: If you ditched this and instead just inlined `this.props.siteId` below, you could collapse the method definition, using some arrow fun:

```js
handleSiteTypeSelection = siteType => () => {
	this.props.saveJetpackOnboardingSettings( this.props.siteId, {
		siteType,
	} );
}
```

Comment (tyxla) [reply client/jetpack-onboarding/steps/site-type.jsx]: Nice, addressed in ec21a1d2e6.

**Comment by tyxla** [comment]
> I'm just wondering -- do we even need to store the site type on the backend? Isn't this step simply relevant for the UI, to decide what the next step is?

TLDR: yes - it would be useful to store it for cases when the user doesn't finish the onboarding flow in one breath, but returns later to complete it. Also, it seems the original plugin stores the site type, so we're only mirroring that functionality: https://github.com/Automattic/jetpack-onboarding/blob/master/class.jetpack-onboarding-end-points.php#L373

**Comment by tyxla** [comment]
@ockham will merge this one after https://github.com/Automattic/jetpack/pull/8431 lands. Let me know if you have any other suggestions before we 🚢 .
